halo/attest/keeper/keeper.go

```diff
@@ -522,6 +503,41 @@ func (k *Keeper) earliestAttestation(ctx context.Context, version xchain.ChainVe
 	return att, true, nil
 }
 
+// listAllAttestations returns all approved attestations for the given chain.
+func (k *Keeper) listAllAttestations(ctx context.Context, version xchain.ChainVersion, status Status, blockOffset uint64) ([]*types.Attestation, error) {
+	defer latency("list_all_attestations")()
+
+	idx := AttestationStatusChainIdConfLevelBlockOffsetIndexKey{}.WithStatusChainIdConfLevelBlockOffset(uint32(status), version.ID, uint32(version.ConfLevel), blockOffset)
+	iter, err := k.attTable.List(ctx, idx)
+	if err != nil {
+		return nil, errors.Wrap(err, "list")
+	}
+	defer iter.Close()
+
+	var resp []*types.Attestation
+	next := blockOffset
+	for iter.Next() {
+		att, err := iter.Value()
+		if err != nil {
+			return nil, errors.Wrap(err, "value")
+		}
+
+		if att.GetBlockOffset() != next {
```

Comment:
I copy-pasta-ed this from `listAttestationsFrom` above, but I'm actually not sure why we break here instead of failing?